CONTRIBUTING.md

```diff
@@ -47,6 +47,61 @@ and additionally the following rules:
 * Use `my ($self) = @_;` for parameter parsing in methods when accessing the
   `$self` object. Do not parse any parameter if you do not need any.
 * [DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself)
+* Avoid `sleep()`: Do not use `sleep()` or simulate a sleep-like functionality.
+  This is introducing the risk of either unnecessarily wasting time or just
+  shifting race conditions and making them even harder to investigate. You *can*
+  use sleep but there are only very limited cases where is the best solution
+  (e.g. yield threads in low-level system code). Here you have the classical
+  problem in software design of
+  [Synchronization](https://en.wikipedia.org/wiki/Synchronization_(computer_science))
+  . sleep should not be used in place of proper synchronization methods.
+  https://blogs.msmvps.com/peterritchie/2007/04/26/thread-sleep-is-a-sign-of-a-poorly-designed-program/
+  is one of multiple references that explains this: "The original problem is
+  likely a timing/synchronization issue, ignoring it by hiding it with
+  Thread.Sleep is only going to delay the problem and make it occur in random,
+  hard to reproduce ways." And if the problem does not occur at least time is
+  wasted.
+  We are waiting because some condition changes some time ... keyword(s) is/are
+  *some time*. It is important to know the exact condition that the code should
+  wait for and check for that instead of delaying program execution for an
+  arbitrary amount of time. For openQA tests in general just try to do what you
+  as a human user would also do. You would not look on your watch waiting for
+  exactly 100s before you look at the screen again, right? :)
+* Avoid `is_tumbleweed()`: Our tests should always consider openSUSE Tumbleweed
+  as the default product without a version. That means any differing behaviour
+  should have an explicit exclude rule for the "older products" and potentially
+  an explanation why the behaviour should differ. You can try to negate the logic
+  check.
+* Consider a corresponding Tumbleweed test: Because of
+  [Factory First](https://opensource.suse.com/suse-open-source-policy)
+  we should test against Tumbleweed by default (with exceptions for special
+  packages, of course). So please check if your code changes work with a current
+  Tumbleweed snapshot and then add to the according schedule for Tumbleweed
```

Comment:
I'd rather add packages and extended testing, unless there's good motive

Reply:
sorry, I don't understand. Do you mean

```suggestion
  we should test against Tumbleweed by default (with exceptions for special
  packages and extended testing, unless there's good motive). So please check if your code changes work with a current
  Tumbleweed snapshot and then add to the according schedule for Tumbleweed
```

?

Because then I would wonder about the "extended testing". It is correct that reality is that the SLE testing scope is much broader than Tumbleweed but this is what I try to cover with "by default". Maybe we can keep this PR as is as baseline considering that the other changes I can already do and as there so many approvals already :) Then you can add a PR and we come up with a better wording for this specific section